modin/pandas/series.py

```diff
@@ -92,7 +93,11 @@ def __init__(
                     )
                 )
             )._query_compiler
-        if len(query_compiler.columns) != 1 or (
+        if shape_hint == "column":
+            assert len(query_compiler.columns) == 1, "wrong shape hint"
+        elif shape_hint == "row":
```

Comment:
The goal is to avoid adding `shape_hint`.

This looks like the main issue for `shape_hint`, where we cannot determine whether or not to `transpose` because the logic for the `if len(...)` is in the API layer. Please correct me if I am wrong here.

The fix that makes the most sense to me is to add a call into the query compiler for something like `convert_to_columnar` (better name suggestion?) and then push this logic down.

Reply:
Yeah, this should help to move shape hint into the backend part, I'll try to move related logic into the query compiler.